src/main/scala/monix/kafka/KafkaConsumer.scala

```diff
@@ -0,0 +1,140 @@
+package monix.kafka
+
+import com.typesafe.scalalogging.StrictLogging
+import monix.execution.Ack.Stop
+import monix.execution.FutureUtils.extensions.FutureExtensions
+import monix.execution.cancelables.SingleAssignmentCancelable
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.{Observable, Observer}
+import monix.reactive.observers.Subscriber
+import org.apache.kafka.clients.consumer.{ConsumerRecord, ConsumerRecords, KafkaConsumer => ApacheKafkaConsumer}
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
+import scala.collection.JavaConverters._
+
+/** An `Observable` implementation that reads messages from `Kafka`.
+  *
+  * You need to call `addTopicSubscription` in order for the Observable to start sending messages from that topic.
+  */
+final class KafkaConsumer[K, V] private
+  (config: KafkaConsumerConfig, ioScheduler: Scheduler)
+  (implicit K: Deserializer[K], V: Deserializer[V])
+  extends Observable[ConsumerRecord[K, V]] with StrictLogging {
+
+  def unsafeSubscribeFn(subscriber: Subscriber[ConsumerRecord[K, V]]): Cancelable = {
+    val source = underlying.onErrorHandleWith { ex =>
+      logger.error(
+        s"Unexpected error in KafkaConsumerObservable, retrying in ${config.retryBackoffTime}", ex)
+      // Applying configured delay before retry!
+      underlying.delaySubscription(config.retryBackoffTime)
+    }
+
+    source.unsafeSubscribeFn(subscriber)
+  }
+
+  def addTopicSubscription(topics: Set[String]): Unit = {
+    logger.info(s"Subscribing to topics : ${topics.mkString(",")}")
+    consumerRef.subscribe((currentSubscriptions() ++ topics).asJavaCollection)
+  }
```

Comment:
I would rather not have such utilities on this class. This class should be a plain Observable in my opinion, with no utilities added. Also this assumes that this class instance has shared state, but it should not share any state, see my comment below for `consumerRef`.

Also notice how you assume that the initialization of `consumerRef` and the `consumerRef.subscribe` calls succeed. That is not true, as from my understanding `consumerRef.subscribe` probably initiates a network connection and if it doesn't, you never know with Java APIs. And if it throws an error, even if you catch it here, you can't do anything with it. Like you can't stream it by means of `onError` because you're not in the context of your observable.